src/components/RadioButton/RadioButton.scss

```diff
@@ -3,45 +3,94 @@
 $block: '.#{variables.$ns}radio-button';
 
 #{$block} {
+    --_--border-width: 1px;
+    --_--transition-time: 0.15s;
+
     box-sizing: border-box;
     display: inline-flex;
     flex-direction: row;
     font-family: var(--g-text-body-font-family);
     font-weight: var(--g-text-body-font-weight);
-    border-radius: var(--_--border-radius);
-    background-color: var(--g-color-base-generic);
     position: relative;
 
-    --_--border-radius-inner: calc(var(--_--border-radius) - 3px);
-
-    &__plate {
-        position: absolute;
-        inset-block: 0;
-        transition:
-            left 0.2s,
-            width 0.2s;
-
-        &[hidden] {
-            display: none;
-        }
-    }
-
     &__option {
+        position: relative;
         flex: 1 1 auto;
         user-select: none;
         font-size: var(--g-text-body-1-font-size);
         text-align: center;
-        border-radius: var(--_--border-radius-inner);
+
         cursor: pointer;
         transform: scale(1);
-        transition: color 0.15s linear;
+        transition: color var(--_--transition-time) linear;
 
-        &-outline {
+        &::before {
+            position: absolute;
+            inset-inline-start: 0;
+            inset-block: var(--_--border-width);
+            content: '';
+            width: var(--_--border-width);
+            background-color: var(--g-color-line-generic);
+        }
+
+        &::after {
             content: '';
             position: absolute;
-            z-index: -1;
-            inset: 3px;
-            border-radius: var(--_--border-radius-inner);
+            z-index: -2;
```

Comment:
In which case is z-index: -1 not sufficient?

Reply:
I forgot to lower it to -1 after testing different layout options.
Fixed it